### PR TITLE
feat: #204 - Add IfStatement.

### PR DESCRIPTION
**Pull request by dicarlo2**
I'll add a `TypeGuard` for `Statement` once the rest are in.

### CODE REVIEW

File: src/utils/TypeGuards.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 /* tslint:disable */
 import * as ts from "typescript";
```
Comment (dsherret) [inline src/utils/TypeGuards.ts]: You can re-generate this file based on your changes by running: `npm run code-generate`.

Sidenote: I'm going to abstract out the duplicate pattern going in in these methods in the next few weeks.

File: src/compiler/statement/IfStatement.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    getThenStatement() {
+        return this.global.compilerFactory.getNodeFromCompilerNode(this.compilerNode.thenStatement, this.sourceFile) as Statement;
+    }
+
+    /**
+     * Gets this if statement's else statement.
+     */
+    getElseStatement() {
+        return this.compilerNode.elseStatement == null
+            ? null
```
Comment (dsherret) [inline src/compiler/statement/IfStatement.ts]: Keep null for the check above, but return `undefined` here. The public api of the library only uses `undefined`.

File: src/compiler/statement/ExpressionStatement.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 ﻿import * as ts from "typescript";
 import {removeStatementedNodeChild} from "./../../manipulation";
 import {Node, Expression} from "./../common";
+import {Statement} from "./Statement"
```
Comment (dsherret) [inline src/compiler/statement/ExpressionStatement.ts]: Semicolon.

File: src/compiler/statement/IfStatement.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import {ChildOrderableNode} from "./../base";
+import {Statement} from "./Statement";
+
+export const IfStatementBase = ChildOrderableNode(Statement);
+export class IfStatement extends IfStatementBase<ts.IfStatement> {
+    /**
+     * Gets this if statement's expression.
+     */
+    getExpression() {
+        return this.global.compilerFactory.getNodeFromCompilerNode(this.compilerNode.expression, this.sourceFile) as Expression;
```
Comment (dsherret) [inline src/compiler/statement/IfStatement.ts]: Sidenote: I really need to just add an internal convenience method:

```ts
this.getNodeFromCompilerNode(this.compilerNode.expression);
```

I'll open an issue for that and then change the entire code base (opened #206)

**Comment by dsherret** [comment]
Thanks again! I can run a minor release tonight for this change.

**Comment by dicarlo2** [comment]
No worries on running a minor release. I have a lot more coming, so totally up to you.

**Comment by dicarlo2** [comment]
And thanks for the quick review!

**Pull request merged**